pkg/operator/configobservation/node/observe_authorization_mode.go

```diff
@@ -0,0 +1,69 @@
+package node
+
+import (
+	"sort"
+
+	"github.com/openshift/api/features"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var defaultAuthenticationModes = []string{
+	"Node",
+	"RBAC",
+	"Scope",
+	"SystemMasters",
+}
+
+type authorizationModeObserver struct {
+	featureGateAccessor featuregates.FeatureGateAccess
+	authModes           []string
+}
+
+func NewAuthorizationModeObserver(featureGateAccessor featuregates.FeatureGateAccess) configobserver.ObserveConfigFunc {
+	return (&authorizationModeObserver{
+		featureGateAccessor: featureGateAccessor,
+	}).ObserveAuthorizationMode
+}
+
+// ObserveAuthorizationMode watches the featuregate configuration and generates the apiServerArguments.authorization-mode
+// It currently hardcodes the default set and adds MinimumKubeletVersion if the feature is set to on.
+func (o *authorizationModeObserver) ObserveAuthorizationMode(genericListers configobserver.Listers, _ events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	defer func() {
+		// Prune the observed config so that it only contains minimumKubeletVersion field.
+		ret = configobserver.Pruned(ret, authModePath)
+	}()
+
+	ret = map[string]interface{}{}
+	if !o.featureGateAccessor.AreInitialFeatureGatesObserved() {
+		return existingConfig, nil
+	}
+
+	featureGates, err := o.featureGateAccessor.CurrentFeatureGates()
+	if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	if err := AddAuthorizationModes(ret, featureGates.Enabled(features.FeatureGateMinimumKubeletVersion)); err != nil {
+		return existingConfig, append(errs, err)
+	}
+	return ret, nil
+}
+
+// AddAuthorizationModes modifies the passed in config
+// to add the "authorization-mode": "MinimumKubeletVersion" if the feature is on. If it's off, it
+// removes it instead.
+// This function assumes MinimumKubeletVersion auth mode isn't present by default,
+// and should likely be removed when it is.
+func AddAuthorizationModes(observedConfig map[string]interface{}, isMinimumKubeletVersionEnabled bool) error {
```

Comment:
Maybe we could split this method into two, one that accepts the value from the FG and another that adds hardcoded/default values.

Reply:
hm the approach taken for pod security (the other featuregated observer) is to set an invalid config in the default so we know it's overridden. I'll do that instead I think

Reply:
https://github.com/openshift/cluster-kube-apiserver-operator/blob/992114aa1298e9c10d2ae7df3a203a02f55c3d59/bindata/assets/config/defaultconfig.yaml#L17-L22 which is then unconditionally overridden by https://github.com/openshift/cluster-kube-apiserver-operator/blob/992114aa1298e9c10d2ae7df3a203a02f55c3d59/pkg/cmd/render/render.go#L350-L357

Reply:
I think that the render command creates the bootstrap manifest, and the target controller creates a manifest for each revision. We need to ensure that the target controller also applies the default set of the auth methods (just in case). WDYT ?